### ISSUE TITLE
[Bug] CollectionView crashes on iOS 12.4 for repeated adds

### ISSUE THREAD

**Issue by Dresel** [xamarin/Xamarin.Forms#8672]
# Description

When calling `ObservableCollection.Add` repeatedly for a CollectionView the application crashes on iOS with the following exception:

> CollectionViewAddItemCrashRepo.iOS[602:47143] *** Assertion failure in -[UICollectionView _endItemAnimationsWithInvalidationContext:tentativelyForReordering:animator:], /BuildRoot/Library/Caches/com.apple.xbs/Sources/UIKitCore/UIKit-3698.140/UICollectionView.m:6165
**Foundation.MonoTouchException:** 'Objective-C exception thrown.  Name: NSInternalInconsistencyException Reason: Invalid update: invalid number of items in section 0.  The number of items contained in an existing section after the update (2) must be equal to the number of items contained in that section before the update (2), plus or minus the number of items inserted or deleted from that section (1 inserted, 0 deleted) and plus or minus the number of items moved into or out of that section (0 moved in, 0 moved out).

This happens for version >= `4.3.x` (including `4.4.0.991210-pre2`). Version `4.2.0.910310` is not affected.

### Steps to Reproduce

1. Bind an `ObservableCollection` to a CollectionView
2. Call `ObservableCollection.Add` repeatedly

```csharp
MainThread.BeginInvokeOnMainThread(() =>
{
	for (int i = 0; i < 10; i++)
	{
		Collection.Add(new ItemViewModel() { Id = i.ToString(), Name = $"Item {i}" });
	}
});
```

### Expected Behavior

No crash.

### Actual Behavior

Crash.

### Basic Information

- Version with issue: >= `4.3.x`
- Last known good version: `4.2.0.910310`
- Platform Target Frameworks:
  - iOS:  12.4

### Reproduction Link

git clone `https://github.com/Dresel/CollectionViewAddItemCrashRepo.git`

**Comment by jfversluis** [comment]
Thanks for the report @Dresel!

I've pulled down your code, updated to the latest stable package and although I do see similar output in the console window, the app doesn't crash. To make the issue a bit clearer I have increased the iteration to 100. You can see my attempt below. Am I missing anything?

![Untitled](https://user-images.githubusercontent.com/939291/69640888-73dca900-105f-11ea-944c-371d56a2a01b.gif)

**Comment by Dresel** [comment]
Good catch, the crash happens on a iOS 12.4.2 iPhone 6 device - on the emulator I see the same results you have.

**Comment by Dresel** [comment]
Crash can be reproduced when running emulator on 12.4.

**Comment by jfversluis** [comment]
Thanks for looking into it. When I run it on iOS 12 indeed it crashes.

Interesting, so for iOS 13 it seems like it's just a warning in the console and for iOS 12 it actually crashes. I did see crash messages like this related to `ListView` I think at some point.

Possible duplicates/related #7510, #7393.

In the related issues I see something about this being fixed 4.4-pre but this is not the case for this issue

**Comment by motasemloubani91** [comment]
Any updates regrading this issue, at least give us estimation time. We are waiting this to be fixed to release our app from a week ago!

**Comment by motasemloubani91** [comment]
I have moved to use a third party library to support mutli columns listview, I don't understand how do you publish a control that has not had a basic test!!

**Comment by jsuarezruiz** [comment]
Recently this PR has been merged: https://github.com/xamarin/Xamarin.Forms/pull/7711
I think this issue is the same and should be fixed. In any other case, we will review the feedback to try to cover it as soon as possible.

**Comment by jsuarezruiz** [comment]
After some tests I can confirm that this issue is already fixed.

![issue8672-fixed](https://user-images.githubusercontent.com/6755973/70229849-48407980-1757-11ea-9f8f-073385156b43.gif)

**Comment by samhouts** [comment]
closed by #7711
